lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControlModule.java

```diff
@@ -15,43 +15,55 @@
 
 import com.google.inject.Binder;
 import com.google.inject.Inject;
-import com.google.inject.Module;
 import com.google.inject.Provides;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 import io.airlift.log.Logger;
 import io.trino.plugin.base.CatalogName;
 import io.trino.spi.connector.ConnectorAccessControl;
 
-import java.io.File;
+import java.util.function.Supplier;
 
 import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
 import static io.airlift.configuration.ConfigBinder.configBinder;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public class FileBasedAccessControlModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     private static final Logger log = Logger.get(FileBasedAccessControlModule.class);
 
     @Override
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
         configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
+        install(conditionalModule(
```

Comment:
Can we use this api - conditionalModule(Class<T> config, Predicate<T> predicate, Module module, Module otherwise)

In future if we have plans on fetching from other endpoint - we could extract them separately.

Reply:
done